src/gui/widgets/TempoSyncBarModelEditor.cpp

```diff
@@ -0,0 +1,315 @@
+}
+
+
+
+
+TempoSyncBarModelEditor::~TempoSyncBarModelEditor()
+{
+	if( m_custom )
+	{
+		delete m_custom->parentWidget();
+	}
+}
+
+
+
+
+void TempoSyncBarModelEditor::modelChanged()
+{
+	if( model() == nullptr )
+	{
+		qWarning( "no TempoSyncKnobModel has been set!" );
+	}
+	if( m_custom != nullptr )
+	{
+		m_custom->setModel( &model()->getCustomMeterModel() );
+	}
+	connect(model(), &TempoSyncKnobModel::syncModeChanged, this, &TempoSyncBarModelEditor::updateDescAndIcon);
+	connect( this, SIGNAL(sliderMoved(float)),
+			model(), SLOT(disableSync()));
+	updateDescAndIcon();
+}
+
+
+
+
+void TempoSyncBarModelEditor::contextMenuEvent( QContextMenuEvent * )
+{
+	mouseReleaseEvent( nullptr );
+
+	CaptionMenu contextMenu( model()->displayName(), this );
+	addDefaultActions( &contextMenu );
+	contextMenu.addSeparator();
+
+	float limit = 60000.0f / ( Engine::getSong()->getTempo() *
+							model()->scale() );
+
+	QMenu * syncMenu = contextMenu.addMenu( m_tempoSyncIcon,
+						m_tempoSyncDescription );
+	if( limit / 8.0f <= model()->maxValue() )
+	{
+
+	connect( syncMenu, SIGNAL(triggered(QAction*)),
+			model(), SLOT(setTempoSync(QAction*)));
+	syncMenu->addAction( embed::getIconPixmap( "note_none" ),
+		tr( "No Sync" ) )->setData( (int) TempoSyncKnobModel::SyncMode::None );
+	if( limit / 0.125f <= model()->maxValue() )
+	{
+		syncMenu->addAction( embed::getIconPixmap( "note_double_whole" ),
+				tr( "Eight beats" ) )->setData(
+				(int) TempoSyncKnobModel::SyncMode::DoubleWholeNote );
+	}
+	if( limit / 0.25f <= model()->maxValue() )
+	{
+		syncMenu->addAction( embed::getIconPixmap( "note_whole" ),
+					tr( "Whole note" ) )->setData(
+					(int) TempoSyncKnobModel::SyncMode::WholeNote );
+	}
+	if( limit / 0.5f <= model()->maxValue() )
+	{
+		syncMenu->addAction( embed::getIconPixmap( "note_half" ),
+					tr( "Half note" ) )->setData(
+					(int) TempoSyncKnobModel::SyncMode::HalfNote );
+	}
+	if( limit <= model()->maxValue() )
+	{
+		syncMenu->addAction( embed::getIconPixmap( "note_quarter" ),
+					tr( "Quarter note" ) )->setData(
+				(int) TempoSyncKnobModel::SyncMode::QuarterNote );
+	}
+	if( limit / 2.0f <= model()->maxValue() )
+	{
+		syncMenu->addAction( embed::getIconPixmap( "note_eighth" ),
+					tr( "8th note" ) )->setData(
+				(int) TempoSyncKnobModel::SyncMode::EighthNote );
+	}
+	if( limit / 4.0f <= model()->maxValue() )
+	{
+		syncMenu->addAction( embed::getIconPixmap( "note_sixteenth" ),
+					tr( "16th note" ) )->setData(
+				(int) TempoSyncKnobModel::SyncMode::SixteenthNote );
+	}
+	syncMenu->addAction( embed::getIconPixmap( "note_thirtysecond" ),
+					tr( "32nd note" ) )->setData(
+				(int) TempoSyncKnobModel::SyncMode::ThirtysecondNote );
+	syncMenu->addAction( embed::getIconPixmap( "dont_know" ),
+				tr( "Custom..." ),
+				this, SLOT(showCustom())
+						)->setData(
+					(int) TempoSyncKnobModel::SyncMode::Custom );
+	contextMenu.addSeparator();
+
+	}
+	contextMenu.exec( QCursor::pos() );
+
+	delete syncMenu;
+}
+
+
+
+
```

Comment:
The `complex method` failure is occurring on this function of you look more closely at the message (`src/gui/widgets/TempoSyncBarModelEditor.cpp#L163-L259` notice L163-L259). Might be because of the switch case.

Reply:
Oops, tagged the wrong line. It's supposed to be line 163.